prometheus_client/samples.py

```diff
@@ -36,8 +37,15 @@ def __gt__(self, other):
 # Timestamp can be a float containing a unixtime in seconds,
 # a Timestamp object, or None.
 # Exemplar can be an Exemplar object, or None.
-Sample = namedtuple('Sample', ['name', 'labels', 'value', 'timestamp', 'exemplar'])
-Sample.__new__.__defaults__ = (None, None)
-
-Exemplar = namedtuple('Exemplar', ['labels', 'value', 'timestamp'])
-Exemplar.__new__.__defaults__ = (None,)
+class Exemplar(NamedTuple):
+    labels: Mapping[str, str]
```

Comment:
We should probably use `Dict` here to be consistent with other label sets we use in for example `inc` on a counter. Alternatively we could have a custom type for label sets that is `Dict[str, str]`?

Reply:
I can change it to `Dict` I'm not as familiar with the code so I put `Mapping` just in case there's a different type that act like a dict.

Reply:
Do you think that type will be used frequently? If so, maybe it would be warranted to declare it specifically. For now I just changed it to `Dict[]`

Reply:
I think just a Dict for now makes sense, it would be easiest for users rather than navigating to another type definition.